wallet/src/actors/worker/methods.rs

```diff
@@ -599,23 +599,39 @@ impl Worker {
         Compat01As03::new(f).await
     }
 
-    /// Try to synchronize the information for a wallet to whatever the world state is in a Witnet
-    /// chain.
+    /// Sync wrapper in order to clear transient addresses in case of errors
     pub fn sync(
         &self,
         wallet_id: &str,
         wallet: &types::SessionWallet,
         sink: types::DynamicSink,
         resynchronizing: bool,
     ) -> Result<()> {
-        let limit = i64::from(self.params.node_sync_batch_size);
-
         // Generate transient addresses for sync purposes
         wallet.initialize_transient_addresses(
             self.params.sync_address_batch_length,
             self.params.sync_address_batch_length,
         )?;
 
+        let sync_result = self._sync(wallet_id, wallet, sink, resynchronizing);
+
+        // Clear transient created addresses
+        wallet.clear_transient_addresses()?;
+
+        sync_result
+    }
+
+    /// Try to synchronize the information for a wallet to whatever the world state is in a Witnet
+    /// chain.
+    pub fn _sync(
```

Comment:
In the wallet, functions starting with `_` usually take a `&mut State` parameter. So let's choose a different naming to avoid confusion, and also make this function private.

```suggestion
    fn sync_inner(
```